src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.spring.config;
+
+import java.security.SecureRandom;
+import java.util.Properties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(inMemoryUserDetailsManager());
+    }
+
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .formLogin()
+                .permitAll()
+                .and()
+                .httpBasic()
+                .and()
+                .csrf().disable();
+    }
+
+    @Bean
+    public InMemoryUserDetailsManager inMemoryUserDetailsManager() {
+        Properties users = new Properties();
+        users.put("user@domain.com", getEncoder().encode("1234") + ",ROLE_USER,enabled");
+        return new InMemoryUserDetailsManager(users);
+    }
+
+    @Bean
+    public PasswordEncoder getEncoder() {
+        int strength = 10;
```

Comment:
it should rather be a constant, definitely not a local variable

Reply:
Sure!
Should i rePR?
Another question. Is everything else OK? Especially, handleExceptionInternal() overriding in CustomGlobalExceptionHandler?